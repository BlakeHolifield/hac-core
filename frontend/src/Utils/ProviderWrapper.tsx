import * as React from 'react';

type ProviderWrapperProps = {
  provider: React.Provider<unknown>;
  useValueHook: () => unknown;
  children?: React.ReactNode;
};

const ProviderWrapper: React.FC<ProviderWrapperProps> = ({ provider: Provider, useValueHook, children }) => {
  const value = useValueHook();
  return <Provider value={value}>{children}</Provider>;
};

export default React.memo(ProviderWrapper);
