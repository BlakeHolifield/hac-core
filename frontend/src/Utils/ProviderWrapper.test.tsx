import * as React from 'react';
import { render } from '@testing-library/react';
import ProviderWrapper from './ProviderWrapper';

describe('ProviderWrapper', () => {
  test('should render correctly', () => {
    const useValueHook = jest.fn();
    useValueHook.mockReturnValueOnce(10);
    const container = render(
      <ProviderWrapper provider={React.createContext<unknown>({}).Provider} useValueHook={useValueHook}>
        foo
      </ProviderWrapper>,
    );
    expect(container.baseElement).toMatchSnapshot();
  });
  test('should pass correct value', () => {
    const useValueHook = jest.fn();
    render(
      <ProviderWrapper provider={React.createContext<unknown>({}).Provider} useValueHook={useValueHook}>
        foo
      </ProviderWrapper>,
    );
    expect(useValueHook).toBeCalledTimes(1);
  });
});
