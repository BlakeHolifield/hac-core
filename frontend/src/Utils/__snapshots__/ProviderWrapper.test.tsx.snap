// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderWrapper should render correctly 1`] = `
<body>
  <div>
    foo
  </div>
</body>
`;
